src/components/AddButton/index.jsx

```diff
@@ -1,83 +1,47 @@
-import { ADD_LIST, ADD_CARD } from "../../actions/constants";
-import styles from './AddButton.module.css';
-
-const AddButton = ({ text, type, listId = null }) => {
-  const [condition, setCondition] = useState({ isFormActive: false });
-  const dispatch = useDispatch();
-  const state = useSelector((state) => state.lists);
-
-  const handleAddActivity = () => {
-    const text = condition.text;
-    if (type === "list" && text) {
-      dispatch({ type: ADD_LIST, payload: { text } });
-    }
-    if (type === "card" && text) {
-      console.log(listId, state);
-      dispatch({ type: ADD_CARD, payload: { text, id: listId } });
+import { useInput } from "../../utils";
+import { RenderFormAddButton } from "../RenderFormAddButton";
+import { RenderAddButton } from "../RenderAddButton";
+import { useClick } from "../../utils";
+
+const AddButton = ({
+  text,
+  type,
+  listId = null,
+  placeholder,
+  textBtn,
+  onClick,
+}) => {
+  const handleAddActivity = (text) => {
+    if (typeof onClick === "function") {
+      setIsFormActive(false);
+      onClick(text);
     }
   };
 
-  const onChange = (ev) => {
-    onChangeCondition("text", ev.target.value);
-  };
+  const [formRef, isFormActive, setIsFormActive] = useClick(false);
+  const input = useInput("");
 
-  const renderForm = (type) => {
-    const placeholder =
-      type === "list"
-        ? "Введите заголовок списка"
-        : "Введите заголовок для это карточки";
-
-    const textFormButton = type === "list" ? "список" : "задачу";
-
-    return (
-      <form>
-        <textarea
-          className="form-control"
-          id="form-control"
-          rows="3"
-          placeholder={placeholder}
-          autoFocus
-          onBlur={() => onChangeCondition("isFormActive", false)}
-          value={condition.text}
-          onChange={onChange}
-          style={styles.textarea}
-        />
-        <div className="input-group d-flex align-items-center">
-          <button
-            type="submit"
-            className="btn btn-primary mt-2"
-            onMouseDown={handleAddActivity}
-          >
-            Добавить {textFormButton}
-          </button>
-          <i className="bi bi-x-lg" style={styles.cross} />
-        </div>
-      </form>
-    );
-  };
-
-  const renderButton = (type, text) => {
-    return (
-      <div
-        style={type === "list" ? styles.list : styles.card}
-        onClick={() => onChangeCondition("isFormActive", true)}
-      >
-        <i className="bi bi-plus-lg" />
-        <span className="p-2">Добавить {text}</span>
-      </div>
-    );
+  const onChangeState = (value) => {
+    setIsFormActive(value);
   };
-  function onChangeCondition(field, value) {
-    setCondition({ ...condition, [field]: value });
-  }
 
   return (
-    <div>
-      {condition.isFormActive
-        ? renderForm(type, text, styles)
-        : renderButton(type, text)}
+    <div ref={formRef}>
+      {isFormActive ? (
```

Comment:
another code structure is recommended here (by Alexei)
example: 
    {isFormActive && (
        <RenderFormAddButton
          onChangeState={onChangeState}
          handleAddActivity={handleAddActivity}
          placeholder={placeholder}
          input={input}
          textBtn={textBtn}
        />
      )}
{!isFormActive && (
        <RenderAddButton
          type={type}
          textBtn={textBtn}
          onChangeState={onChangeState}
        />
      )}